perun/logic/index.py

```diff
@@ -580,23 +580,22 @@ def remove_from_index(base_dir, minor_version, removed_file_generator, remove_al
         all_entries.sort(key=lambda unsorted_entry: unsorted_entry.offset)
         removed_entries = []
 
-        for removed_file in removed_file_generator:
+        for i, removed_file in enumerate(removed_file_generator):
             def lookup_function(entry):
                 """Helper lookup function according to the type of the removed file"""
                 if store.is_sha1(removed_file):
                     return entry.checksum == removed_file
                 else:
                     return entry.path == removed_file
 
-            if remove_all:
-                removed_entries.append(
-                    lookup_all_entries_within_index(index_handle, lookup_function)
-                )
-            else:
-                removed_entries.extend([
-                    lookup_entry_within_index(index_handle, lookup_function, removed_file)
-                ])
-            perun_log.info("deregistered: {}".format(removed_file))
+            found_entry = lookup_entry_within_index(index_handle, lookup_function, removed_file)
+            removed_entries.append(found_entry)
+
+            perun_log.info("{}/{} deregistered {} from index".format(
```

Comment:
Good point, I left it like that since I was not sure if it is worth extracting (since we may want different format of output for pending/index). I'll keep it for now as it is, but it may be worthy in future.